packages/venia-ui/lib/components/CartPage/PriceAdjustments/ShippingMethods/shippingMethods.js

```diff
@@ -24,43 +24,46 @@ const ShippingMethods = props => {
 
     const classes = mergeClasses(defaultClasses, props.classes);
 
+    const radios =
+        isShowingForm && hasMethods ? (
```

Comment:
How about using `useMemo` here? @sirugh and I had a conversation yesterday about this. He can expand more on the `useMemo` use case.

Reply:
We have to draw the line somewhere on memoization, and in this codebase, I've generally drawn that line at element creation. The only place we should be memoizing element creation (afaik) is when we're doing it in a loop.

Reply:
> when we're doing it in a loop

Like...a _render_ loop? 😆

Reply:
But seriously, imagine a toggleable element that we expect a user to interact with quite often. Shouldn't we memoize the state to avoid re-creating the component unnecessarily?

This is the difference we're talking about, I think. Which of these is "more performant"/"better"?
```js
const MyComponent = ({ renderChildren }) => {
    const FooComponent = useMemo(() => {
        return renderChildren ? (
            <Foo />
        ) : null;
    });

    const BarComponent = renderChildren ? (
        <Bar />
    ) : null

    return (
        <Fragment>
            {FooComponent}
            {BarComponent}
        </Fragment>
    )
}
```

Reply:
Yeah, I feel the same. `useMemo` can save on garbage collection if we are smart about the dependency array element. `useMemo` can quickly go out of hand if we abuse it by putting too many elements in the dependency array or elements that have a wide range of possible values. I don't think there is an eslint rule that can help us in this case but I feel `useMemo` has more potential that we are not using at this point in our code base.

Reply:
> This is the difference we're talking about, I think. Which of these is "more performant"/"better"?

> I feel `useMemo` has more potential that we are not using at this point in our code base.

Yes, I understand. The difference here is negligible—it's one `createElement()` call. The benefit is not zero, but it's insignificant.

```jsx
const MyComponent = props => {
  // this is also a `createElement` call
  // should we memoize it too?
  return <div>{props.text}</div>
}
```

If we wanted to realize the full potential of memoization, we could reproduce `PureComponent`.

```jsx
// let's create a custom hook
// it'll only re-render when props change
const usePureComponent = (Component, props) => useMemo(
  () => <Component {...props} />
  [props]
)

// looks pure to me
const MyComponent = props => {
  return <div>{props.text}</div>
}

// neat
export default const pure = props => usePureComponent(MyComponent, props)
```

This is cool, but it's overkill. We're not gaining much from the memoization, and now we're doing something that's pretty unusual. So this is where we draw the line.

We use it to avoid identities changing across renders, and as an early optimization when mapping props to elements in a loop, but not for literally every `createElement` call that happens to be pure (suitable for memoization).

Reply:
> But seriously, imagine a toggleable element that we expect a user to interact with quite often. Shouldn't we memoize the state to avoid re-creating the component unnecessarily?

Situationally, any time you have a particularly complex tree, you can consider memoizing it to improve the render performance. Toggling doesn't factor into it much, though.

Reply:
A question @revanth0212 had in the office was why not just memoize all the time since you may not know if a tree is complex or not.